mscore/musescore.cpp

```diff
@@ -4948,13 +4955,28 @@ int main(int argc, char* av[])
             QString temp = parser.value("T");
             if (temp.isEmpty())
                    parser.showHelp(EXIT_FAILURE);
-            trimMargin = temp.toInt(); // ToDo: check whether conversion went OK, set to -1 if not
+            bool ok = false;
+            trimMargin = temp.toInt(&ok);
+            if (!ok)
+                  trimMargin = -1;
            }
       if (parser.isSet("x")) {
             QString temp = parser.value("x");
             if (temp.isEmpty())
                    parser.showHelp(EXIT_FAILURE);
-            guiScaling = temp.toDouble(); // ToDo: check whether conversion went OK, set to 1.0 if not
+            bool ok = false;
+            guiScaling = temp.toDouble(&ok);
+            if (!ok)
+                  guiScaling = 0.0;
```

Comment:
-x was implemented long before we were using Qt 5.6, so I don't really know anything about the support Qt now provides.  I guess this has to do with "device pixel ratio"?  But indeed, -x should almost never be needed any more, because getting a good physical dots per inch value from Qt allows us to calculate our own guiScaling factor.  And in the cases where Qt is reporting a bad physical dots per inch, it's normally -D you'll want to use as an override, not -x.  The difference being, is the pysical dots per inch is off, this affects score size as well as icon sizes.  So -D fixes both at once.  -x currently says, keep the physical dots per inch value for use in determining score size, but scale icons and other gui elements.  That's different from 2.0.2 where it actually controlled both, but to me, the current way is better, because it actually gives you independent control.  We do of course also provide icon size option preferences, but that doesn't affect palette icons.

So anyhow I do like the way these options work from a user perspective.  But the *implementation* is indeed way more hacky than it should be.  If there is a way to tell Qt 5.6 to scale icons with needing to constantly apply guiScaling manually all over the place, though, I'm all for that.